### PR TITLE
authors: profile page

### PR DESCRIPTION
**Pull request by MSusik**
Signed-off-by: Mateusz Susik <mateusz.susik@cern.ch>

The profiles don't work without mockup data provided. The purpose of this PR is to split the code into few PR, so the integration will be easier.

If you want some demo data, ping me, as I didn't want to include it inside the code.

The PR includes few `Jasmine` tests.

TO DO:
- [ ] replace image with our own.
- [ ] change wording
- [ ] implement everything that is mentioned in `__init__.py` file (not a purpose of this PR).

### CODE REVIEW

**Comment by MSusik** [comment]
@glouppe @jmartinm

File: inspire/modules/authors/static/js/authors/templates/statstable.mustache [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  INSPIRE is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+  59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+}}
+{{#stats}}<tr><th>&emsp;{{#translate}}{{name}}{{/translate}}</th><td><a href={{link}}>{{value}}</a></td></tr>{{/stats}}
```
Comment (glouppe) [inline inspire/modules/authors/static/js/authors/templates/statstable.mustache]: Missing double quotes in the href

Comment (glouppe) [inline inspire/modules/authors/static/js/authors/templates/statstable.mustache]: And shall we use this `&emsp;` space? (cosmetic aspects should only be set through CSS)

File: inspire/modules/authors/templates/authors/author_stub.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#}
+
+{%- extends 'page.html' -%}
+
+{%- block body -%}
+  <div class="alert alert-warning">
+    {{ _("You tried to access an author module page.
+    Author module is currently disabled. A new version
+    is under development and will be released soon.") }}
```
Comment (glouppe) [inline inspire/modules/authors/templates/authors/author_stub.html]: Just wondering, what is the difference between `_( ... )` and `{{#translate}}` (used above)?

File: inspire/modules/authors/templates/authors/personal_details.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      {{ personal_details_row( _('Field'), _('Fields'), record, ('field_categories', 'name')) }}
+      {{ personal_details_row( _('Affiliation'), _('Affiliations'), record, ('positions_list',)) }}
+      {% for id in record.ids %}
+        {% if id.type == 'orcid' %}
+        <tr>
+          <th>{{ _('Identifiers') }}</th>
+          <td>ORCID: <a href='http://www.orcid.org/{{ id.value }}'>{{ id.value }}</a></td>
+        </tr>
+        {% endif %}
+      {% endfor %}
```
Comment (glouppe) [inline inspire/modules/authors/templates/authors/personal_details.html]: Could you also add `BAI`? I think these will stay for some time...

File: inspire/modules/authors/templates/authors/profile.html [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+## along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#}
+{%- extends "page.html" -%}
+
+<!-- TO DO: move toggles to another template/js file -->
+{%- macro render_toggle(name, text) -%}
+<li>
+  <input class="profile-toggle" type="checkbox" name="{{ name }}-checkbox"
+  checked data-size="mini">&emsp;{{ text }}
```
Comment (glouppe) [inline inspire/modules/authors/templates/authors/profile.html]: I think the proper syntax is `checked="checked"` (even though `checked` is supported by most browsers)

Comment (kaplun) [inline inspire/modules/authors/templates/authors/profile.html]: `checked` is valid HTML5, which I believe is what we specify nowadays as doctype.

File: inspire/modules/authors/templates/authors/author_stub.html (hunk 2 above)
Comment (MSusik) [inline inspire/modules/authors/templates/authors/author_stub.html]: In `mustache` there is no `_()` method, so I defined my own `{{#translate}}`. It needs still some work, though.

File: inspire/modules/authors/static/js/authors/templates/statstable.mustache (hunk 1 above)
Comment (MSusik) [inline inspire/modules/authors/static/js/authors/templates/statstable.mustache]: I will add double quotes after the second comment is resolved, so not to hide it.

File: inspire/modules/authors/templates/authors/personal_details.html (hunk 3 above)
Comment (MSusik) [inline inspire/modules/authors/templates/authors/personal_details.html]: BTW, how do we want to represent authors in `pu`. How will the identifiers look?

File: inspire/modules/authors/templates/authors/profile.html (hunk 4 above)
Comment (MSusik) [inline inspire/modules/authors/templates/authors/profile.html]: It is a good point, and I will make it a bit more general.

Can I assume that I can use `ECMAScript5` and `HTML5`?

File: inspire/modules/authors/templates/authors/author_stub.html (hunk 2 above)
Comment (jalavik) [inline inspire/modules/authors/templates/authors/author_stub.html]: Forgive my ignorance, but would it work to have the data passed to `mustache` already translated from the server?

Comment (MSusik) [inline inspire/modules/authors/templates/authors/author_stub.html]: Fair point.

Unfortunately, it doesn't work, and I will look for a solution in the future.

**Comment by jalavik** [comment]
Ping. Is this stalled or awaiting integration etc.?

**Comment by glouppe** [comment]
@jmartinm was making a few changes as far as I remember. In any case, dependencies have now been all merged in. This one is good to go as well. We can still improve it over time. Hence, +1 for merge on my side.

@MSusik Could you rebase?

**Comment by MSusik** [comment]
Firstly, I would like @jmartinm or other developers to rewiev the code.

File: inspire/modules/authors/views.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+blueprint = Blueprint('authors', __name__, url_prefix="/authors",
+                      template_folder='templates',
+                      static_folder='static')
+
+
+@blueprint.route('/profile/<cname>')
+def profile(cname):
+    """Profile."""
+
+    record = {}
```
Comment (glouppe) [inline inspire/modules/authors/views.py]: Now that the author model has been merged in, could you fetch the actual author record? Soon @etzemis will provide fixtures to have examples.

File: inspire/modules/authors/views.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import json
+
+from flask import Blueprint, render_template
+
+blueprint = Blueprint('authors', __name__, url_prefix="/authors",
+                      template_folder='templates',
+                      static_folder='static')
+
+
+@blueprint.route('/profile/<cname>')
```
Comment (glouppe) [inline inspire/modules/authors/views.py]: We should also enable to fetch author by author record id.

File: inspire/modules/authors/views.py (hunk 5 above)
Comment (MSusik) [inline inspire/modules/authors/views.py]: Not really. The profile page will use a lot computed fields. Those will not be ready until `relations` module is ready. Without those fields, the profile page won't render anything.

File: inspire/modules/authors/views.py (hunk 6 above)
Comment (MSusik) [inline inspire/modules/authors/views.py]: Should it be accessible then by `/author/profile/<id>' or '/record/<id>' ?

**Comment by jacenkow** [comment]
@jmartinm I think we can close it.

**Comment by kaplun** [comment]
This is superseded by #990

**Pull request closed without merging**